fedimint-server/src/config/api.rs

```diff
@@ -755,31 +811,44 @@ mod tests {
                 p2p_url,
                 api_url: api_url.clone(),
             };
-            let server = run_server(
-                data_dir.clone(),
-                our_connections.clone(),
-                db,
-                Default::default(),
-                Default::default(),
-                Default::default(),
-            )
-            .await;
+            let dir = data_dir.join(name_suffix.to_string());
+            fs::create_dir_all(dir.clone()).expect("Unable to create test dir");
+
+            let connections = our_connections.clone();
+            let mut task_group = TaskGroup::new();
+            let subgroup = task_group.make_subgroup().await;
+            task_group
+                .spawn("run-server", move |_| async move {
+                    run_server(
+                        dir,
+                        connections,
+                        db,
+                        Default::default(),
+                        Default::default(),
+                        Default::default(),
+                        subgroup,
+                    )
+                    .await
+                })
+                .await;
             // our id doesn't really exist at this point
             let auth = ApiAuth(format!("password-{port}"));
-            let client = WsAdminClient::new(api_url.clone(), PeerId::from(0), auth.clone());
+            let client = WsAdminClient::new(api_url.clone(), PeerId::from(0), auth);
 
             TestConfigApi {
                 client,
-                server,
-                auth,
+                task_group,
                 name,
                 our_connections,
             }
         }
 
-        /// Helper function using the auth we generated
-        async fn set_password(&self) -> FederationResult<()> {
-            self.client.set_password(self.auth.clone()).await
+        /// Helper function using the auth we generated, retries until the API
+        /// is up
+        async fn retry_set_password(&self) {
+            while self.client.set_password().await.is_err() {
+                sleep(Duration::from_millis(100)).await;
```

Comment:
`info!("stuff not ready");` would give some indication when things went wrong and this is stuck here.